### PR TITLE
Fixes #7779 - Default Browser Onboarding Card

### LINKED ISSUE #7779: FXIOS-1337 ⁃ Default Browser Onboarding Card
**Vision**

When I first install Firefox, I want help getting it set up as my default browser, so that I can use the same browser whether I'm opening it from the home screen or from a link in another app

**Acceptance Criteria**

* Default browser card shows up on 3rd run for new users if they have not clicked "learn more" in the new tab default browser message
* This replaces the current ETP card
* Probes will be added in FXIOS-1422

**Notes**

* [Specs](https://www.figma.com/file/3ICmMVvNYMbPV0gfVWCij5/Default-Browser?node-id=202%3A21)

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-1337)

### PR DESCRIPTION
**Pull request by nbhasin2**

### CODE REVIEW

**Comment by nbhasin2** [comment]
cc @dnarcese draft pr for future work

**Comment by isabelrios** [comment]
If we are modifying the Onboarding we may need to update some UI/XCUITests too, I will look at that...

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -622,7 +622,8 @@ class BrowserViewController: UIViewController {
 
     override func viewDidAppear(_ animated: Bool) {
         presentIntroViewController()
-        presentETPCoverSheetViewController()
+        // presentETPCoverSheetViewController()  Hiding ETP onboarding for DB onboarding
```
Comment (nbhasin2) [inline Client/Frontend/Browser/BrowserViewController.swift:625]:
```suggestion
```

File: Client/Frontend/DefaultBrowserOnboarding/DefaultBrowserOnboardingViewController.swift [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        // Constraints
+        setupView()
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+    }
+    
+    private func setupView() {
+        // Done button target setup
```
Comment (nbhasin2) [inline Client/Frontend/DefaultBrowserOnboarding/DefaultBrowserOnboardingViewController.swift:160]:
```suggestion
        // Close button target setup
```

File: Client/Frontend/DefaultBrowserOnboarding/DefaultBrowserOnboardingViewController.swift [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+        closeButton.addTarget(self, action: #selector(dismissAnimated), for: .touchUpInside)
+        // Done button constraints setup
```
Comment (nbhasin2) [inline Client/Frontend/DefaultBrowserOnboarding/DefaultBrowserOnboardingViewController.swift:162]:
```suggestion
        // Close button constraints setup
```

**Comment by dnarcese** [comment]
Simulator Screen Shot - iPhone 12 mini - 2021-02-11 at 16 02 54
Simulator Screen Shot - iPhone 12 Pro Max - 2021-02-11 at 16 05 09
Simulator Screen Shot - iPad Air (4th generation) - 2021-02-11 at 16 14 42

**Comment by nbhasin2** [comment]
@dnarcese made some more changes and here is how it looks like on simulator side by side, let me know what you think

Screen Shot 2021-02-11 at 8 11 11 PM

- Fixed rotation to portrait
- Made the view respect iPhone SE height as it was broken on that
- Made the text part of the view to be more in centre

**Comment by dnarcese** [comment]
@nbhasin2 looks good, thanks!

@brampitoyo can we get a signoff from you based on Nish's screenshots?

**Review by dnarcese — APPROVED** [review]

**Comment by brampitoyo** [comment]
@nbhasin2 All the iPhone screenshots look great.

For the iPad ones, can we follow the dimension, positioning/margin/padding and text style of either the Welcome/Whats’s New Dialog?

- [Figma - Light Theme](https://www.figma.com/file/3ICmMVvNYMbPV0gfVWCij5/Default-Browser?node-id=453%3A4758)
- [Figma - Dark Theme](https://www.figma.com/file/3ICmMVvNYMbPV0gfVWCij5/Default-Browser?node-id=453%3A4758)

Example from Find My app:

whats-new-landscape@2x
whats-new-portrait@2x

**Comment by dnarcese** [comment]
@brampitoyo How's this?
Simulator Screen Shot - iPad Pro (12 9-inch) (4th generation) - 2021-02-16 at 14 56 02

**Review by brampitoyo — APPROVED** [review]
Perfect!

**Comment by isabelrios** [comment]
@dnarcese I checked the latest Bitrise build and now the XCUITest that was failing is working, it [passed](https://addons-testing.bitrise.io/builds/fb2ff0020cc21490/testreport/6e5e2c7c-21f9-40e2-b48f-7e4c41717c44/testsuite/0/testcases?status=failed). But the build is still red, there is now a compilation issue with the Unit Tests:
```
▸ Compiling DefaultBrowserOnboardingTests.swift
❌  /Users/vagrant/git/ClientTests/DefaultBrowserOnboardingTests/DefaultBrowserOnboardingTests.swift:27:50: type 'PrefsKeys' has no member 'KeyDefaultBrowserCardSessionCount'
        sessionValue = prefs.intForKey(PrefsKeys.KeyDefaultBrowserCardSessionCount) ?? 0
                 ^
❌  /Users/vagrant/git/ClientTests/DefaultBrowserOnboardingTests/DefaultBrowserOnboardingTests.swift:39:50: type 'PrefsKeys' has no member 'KeyDefaultBrowserCardSessionCount'
        sessionValue = prefs.intForKey(PrefsKeys.KeyDefaultBrowserCardSessionCount) ?? 0
```

**Pull request merged**